### PR TITLE
Userpilot destination

### PR DESCRIPTION
**Pull request by ThabetIbrahim**
_A summary of your pull request, including the what change you're making and why._

Adding support for Userpilot cloud action destination.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by joe-ayoub-segment** [comment]
Hi again @ThabetIbrahim , 
If you grant me permission to your repo I can quickly make the changes myself. Let me know. 
Best regards,
Joe

**Comment by ThabetIbrahim** [comment]
Thank you for the help @joe-ayoub-segment, I sent you an invite for the repo.

Based on my notes and the ones you shared, I believe the only missing thing I opted not to add is the `createdAt` field in the `identifyUser/index.ts` file. Do you know if we need anything else here?

File: packages/destination-actions/src/destinations/userpilot/identifyUser/index.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    },
+    anonymousId: {
+      type: 'string',
+      required: false,
+      description: 'Anonymous user ID.',
+      label: 'Anonymous ID',
+      default: {
+        '@path': '$.anonymousId'
+      }
+    },
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/userpilot/identifyUser/index.ts:30]: Hi @ThabetIbrahim , 

I'm thinking we should do the same as with the browser destination and just have a single field to collect the main identifier for the user. 

Something like this: 

    userId: {
      type: 'string',
      required: false,
      description: 'The ID of the logged-in user.',
      label: 'User ID',
      default: {
        '@if': {
          exists: { '@path': '$.userId' },
          then: { '@path': '$.userId' },
          else: { '@path': '$.anonymousId' }
        }
      }
    },

File: packages/destination-actions/src/destinations/userpilot/trackEvent/index.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    },
+    anonymousId: {
+      type: 'string',
+      required: false,
+      description: 'Anonymous user ID.',
+      label: 'Anonymous ID',
+      default: {
+        '@path': '$.anonymousId'
+      }
+    },
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/userpilot/trackEvent/index.ts:30]: Hi again @ThabetIbrahim , as per the identifyUser Action and the web Destination, I think we should have a single identifier for the main user id. As below. 

    userId: {
      type: 'string',
      required: false,
      description: 'The ID of the logged-in user.',
      label: 'User ID',
      default: {
        '@if': {
          exists: { '@path': '$.userId' },
          then: { '@path': '$.userId' },
          else: { '@path': '$.anonymousId' }
        }
      }
    },

File: packages/destination-actions/src/destinations/userpilot/identifyUser/index.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    traits: {
+      type: 'object',
+      required: false,
+      description: 'Segment traits',
+      label: 'Traits',
+      default: {
+        '@path': '$.traits'
+      }
+    }
+  },
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/userpilot/identifyUser/index.ts:40]: Hi @ThabetIbrahim , 
Could we add the createdAt field here as well? 
Should we not make the fields as similar between the Deivce Mode and Cloud Mode Destinations?
Keen to hear your thoughts.

File: packages/destination-actions/src/destinations/userpilot/index.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import type { Settings } from './generated-types'
+
+import { getDeleteRequestParams, getValidationParams } from './request-utils'
+
+import identifyUser from './identifyUser'
+
+import trackEvent from './trackEvent'
+
+const destination: DestinationDefinition<Settings> = {
+  name: 'Userpilot Cloud Mode (Actions)',
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/userpilot/index.ts:13]:
```suggestion
  name: 'Userpilot Cloud (Actions)',
```

**Review by joe-ayoub-segment — CHANGES_REQUESTED** [review]
Hi @Thabetlbrahim, 
Main comments are around making this Integration's fields be as similar to the Device Mode Integration's fields, if possible. 

Are you OK making these changes?
Best regards,
Joe

File: packages/destination-actions/src/destinations/userpilot/identifyUser/index.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  perform: (request, { settings, payload }) => {
+    const { traits, userId, createdAt } = payload
+
+    traits?.createdAt && delete traits.createdAt
+
+    const { url, options } = getIdentifyRequestParams(settings, {
+      traits: { ...traits, created_at: createdAt || traits?.created_at },
```
Comment (ThabetIbrahim) [inline packages/destination-actions/src/destinations/userpilot/identifyUser/index.ts:51]: Hi @joe-ayoub-segment, I noticed a behavior where the `created_at` property would be dropped if it's being passed directly without passing the `createdAt` as well.

File: packages/destination-actions/src/destinations/userpilot/identifyUser/index.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+const action: ActionDefinition<Settings, Payload> = {
+  title: 'Identify User',
+  description:
+    'Defines a user in Userpilot, you can visit [Userpilot docs](https://docs.userpilot.com/article/23-identify-users-track-custom-events) for more information.',
+  defaultSubscription: 'type = "identify"',
+  fields: {
+    userId: {
+      type: 'string',
+      required: false,
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/userpilot/identifyUser/index.ts:15]:
```suggestion
      required: true,
```

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
Hi @ThabetIbrahim this PR has been deployed. I'll email you instructions for testing.

**Pull request merged**